packages/babel-plugin-transform-es2015-parameters/test/fixtures/parameters/rest-member-expression-optimisation/expected.js

```diff
@@ -1,9 +1,9 @@
 var t = function () {
-    var x = arguments[0];
-    var y = arguments[1];
+    var x = arguments.length <= 0 || arguments[0] === undefined ? undefined : arguments[0];
```

Comment:
Good catch! The reason behind this code is [this code](https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-es2015-parameters/src/default.js#L8-L9). But you're absolutely right, it's uselessly complicated here.